packages/graph-explorer-proxy-server/node-server.js

```diff
@@ -174,16 +174,59 @@ async function fetchData(res, next, url, options, isIamEnabled, region, serviceT
   }
   // POST endpoint for SPARQL queries.
   app.post("/sparql", async (req, res, next) => {
+    // Gather info from the headers
+    const queryId = req.headers["queryid"];
+    const graphDbConnectionUrl = req.headers["graph-db-connection-url"];
+    const isIamEnabled = !!req.headers["aws-neptune-region"];
+    const region = isIamEnabled ? req.headers["aws-neptune-region"] : "";
+    const serviceType = isIamEnabled ? (req.headers["service-type"] ?? DEFAULT_SERVICE_TYPE) : "";
+
+    /// Function to cancel long running queries if the client disappears before completion
+    async function cancelQuery() {
+      if (!queryId) {
+        return;
+      }
+      proxyLogger.debug(`Cancelling request ${queryId}...`);
+      return await retryFetch(
+        new URL(`${graphDbConnectionUrl}/sparql/status`),
```

Comment:
This will only work for Neptune, we should, at least, add another ticket to update the code where needed to support other SPARQL endpoints. 
The is no standardised way to cancel a connection to a SPARQL endpoint, so we will need a switch in future (neptune / generic-sparql) to only have certain features (like this one) when neptune is selected.

Reply:
I've added issue #271 to cover this.